cms/static_src/javascript/auth.js

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable no-console */
+
+import SessionManagement from 'dis-authorisation-client-js';
+
+/** ----------------------------------------------------------------
+ *  Grab the JSON produced by:
+ *      {{ AUTH_CONFIG | json_script('auth-config') }}
+ *  in the Jinja template (or the Python hook).
+ * ----------------------------------------------------------------*/
+
+let authConfig = {};
+const configEl = document.getElementById('auth-config');
+
+if (configEl) {
+  try {
+    authConfig = JSON.parse(configEl.textContent);
+  } catch (e) {
+    console.error('[WAGTAIL] Failed to parse <script id="auth-config">:', e);
+  }
+}
+
+const {
+  wagtailAdminHomePath,
+  csrfCookieName,
+  csrfHeaderName,
+  sessionRenewalOffsetSeconds,
+  authTokenRefreshUrl,
+  idTokenCookieName,
+} = authConfig;
+
+const { origin } = window.location;
+const logoutURL = `${origin}/${wagtailAdminHomePath}logout/`;
+const extendSessionURL = `${origin}/${wagtailAdminHomePath}extend-session/`;
+
+function getCookieByName(name) {
+  console.debug('[WAGTAIL] Getting cookie by name:', name);
+  const cookies = document.cookie;
+  if (!cookies) {
+    return null;
+  }
+  const cookie = cookies
```

Comment:
Suggestion: Because of the sensitivity of these cookies, ideally we'd want to set them `httpOnly`, so this would stop working.

For CSRF, this can be obtained with `window.wagtailConfig.CSRF_TOKEN`

The ID token is _probably_ fine to keep accessible from JS, but ideally we could pull that from something in the request body instead. Given this is loaded on the page, there might be something in Wagtail already.